### PR TITLE
PHP: Adds "finally", "fn", "match", and "yield" to reserved names

### PR DESCRIPTION
**Pull request by bshaffer**
release notes: yes

Updates the list of reserved names to be consistent with the [official PHP documentation](https://www.php.net/manual/en/reserved.other-reserved-words.php)

Edit: Updated with "match" for PHP 8.0 and "fn" for PHP 7.4

### CODE REVIEW

**Comment by TeBoring** [comment]
https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/compiler/php/php_generator.cc#L60

Also here

**Comment by bshaffer** [comment]
@TeBoring thank you, done!

**Comment by bshaffer** [comment]
@TeBoring ping on this, is anything needed to get this merged?

File: php/ext/google/protobuf/protobuf.c [hunk 1]
```diff
@@ -270,21 +270,21 @@ bool is_inited_file_wrappers;
...
+    "finally",    "for",        "foreach",     "function",     "global",
+    "goto",       "if",         "implements",  "include",      "include_once", 
+    "instanceof", "insteadof",  "interface",   "isset",        "list",
+    "namespace",  "new",        "or",          "print",        "private",
+    "protected",  "public",     "require",     "require_once", "return",
+    "static",     "switch",     "throw",       "trait",        "try",
+    "unset",      "use",        "var",         "while",        "xor",
+    "yield",      "int",        "float",       "bool",         "string",
+    "true",       "false",      "null",        "void",         "iterable"};
 const int kReservedNamesSize = 73;
```
Comment (TeBoring) [inline php/ext/google/protobuf/protobuf.c]: You also need to change this number

**Comment by TeBoring** [comment]
Could  you also update generated_class_test.php to cover the new reserved words?

**Comment by TeBoring** [comment]
https://github.com/protocolbuffers/protobuf/blob/a104dffcb6b1958a424f5fa6f9e6bdc0ab9b6f9e/php/src/Google/Protobuf/Internal/GPBUtil.php#L290

Also here

**Comment by bshaffer** [comment]
> https://github.com/protocolbuffers/protobuf/blob/a104dffcb6b1958a424f5fa6f9e6bdc0ab9b6f9e/php/src/Google/Protobuf/Internal/GPBUtil.php#L290
> 
> Also here

Done!

**Comment by haberman** [comment]
Hmm, it looks like this PR has fallen out of date, but is still relevant. @bshaffer are you interested in bring it up to date?

**Comment by gwojdaszewicz** [comment]
Hi, anything on this? PHP 8 introduced new reserved keyword "match", can you please add it too?

**Comment by bshaffer** [comment]
@haberman Updated with additional keywords. This is not the most fun PR to create, so PTAL before it goes stale again...

File: php/tests/proto/test_reserved_enum_lower.proto [hunk 2]
```diff
@@ -66,6 +69,7 @@ enum use { ZERO61 = 0; }
 enum var { ZERO62 = 0; }
 enum while { ZERO63 = 0; }
 enum xor { ZERO64 = 0; }
+enum yield { ZERO30 = 0; }
```
[the remark this answers is not in the document]
Comment (bshaffer) [reply php/tests/proto/test_reserved_enum_lower.proto:72]: I must not have saved the file... updated.

**Review by fowles — APPROVED** [review]

**Comment by gwojdaszewicz** [comment]
@bshaffer @fowles guys, bad news, i'm getting seg fault when trying to use protoc built from master branch after this change :see_no_evil: 

> protoc --proto_path=/ --grpc_out=/samples --php_out=/samples --plugin=protoc-gen-grpc=/usr/local/bin/grpc_php_plugin /samples/sample.proto
Segmentation fault (core dumped)

**Comment by bshaffer** [comment]
@gwojdaszewicz I fixed it in #8787 but I have no clue how this got by the tests...

**Comment by TeBoring** [comment]
@bshaffer I think this PR also forgot to add new values to `test_reserved_enum_value_upper.proto`

**Comment by gwojdaszewicz** [comment]
@TeBoring any idea when it will be available on [protocolbuffers/protobuf-php](https://github.com/protocolbuffers/protobuf-php)?

**Pull request merged**